pdf_craft/common/folder.py

```
@@ -2,16 +2,29 @@
 from pathlib import Path
 
 
+def assert_within_root(path: Path, root: Path) -> None:
+    """Raise ValueError if path does not resolve to a location inside root."""
+    try:
+        path.resolve().relative_to(root.resolve())
+    except ValueError:
+        raise ValueError(
+            f"Path '{path}' must be within the safe root '{root}'"
+        )
```

Ruff (0.15.13)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)